include/fiction/technology/sidb_cluster_hierarchy.hpp

```diff
@@ -0,0 +1,1080 @@
+//
+// Created by Willem Lambooy on 04.02.2024.
+//
+
+#ifndef FICTION_SIDB_CLUSTER_HIERARCHY_HPP
+#define FICTION_SIDB_CLUSTER_HIERARCHY_HPP
+
+#if (FICTION_ALGLIB_ENABLED)
+
+/**
+ * Uncomment this line to switch to STL containers, which are slower than their respective analogues from the
+ * Parallel-Hashmap library by Gregory Popovitch, but may be inspected with ease in a debugger.
+ */
+// #define DEBUG_SIDB_CLUSTER_HIERARCHY
+
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/technology/sidb_charge_state.hpp"
+#include "fiction/utils/hash.hpp"
+
+#ifdef DEBUG_SIDB_CLUSTER_HIERARCHY
+#include <set>
+#include <unordered_map>
+#include <unordered_set>
+#else
+#include <btree.h>
+#include <phmap.h>
+#endif
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cstdint>
+#include <initializer_list>
+#include <iterator>
+#include <limits>
+#include <memory>
+#include <utility>
+#include <vector>
```

Comment:
include <set> is missing, right?

Reply:
This include is hidden under de DEBUG macro switch. I replaced all std sets with the phmap alternatives. If I missed one I'd be happy to hear